src/tasks/Common/Utils.cs

```diff
@@ -64,6 +64,9 @@ static string CreateTemporaryBatchFile(string command)
             using StreamWriter sw = new(file);
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
+                // set encoding to UTF-8 -> full Unicode support is needed for usernames -
+                // `command` contains tmp dir path with the username
+                sw.WriteLine("chcp 65001>nul");
```

Comment:
msbuild [Exec task](https://github.com/dotnet/msbuild/blob/47a712160eb420136ec983be5e0f412dc1c25b55/src/Tasks/Exec.cs#L210) uses `%SystemRoot%\System32\chcp.com` because of this potential issue: https://github.com/dotnet/msbuild/commit/fe5cd726095b710b14f99625f6d02ab430d41cad